src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -17,21 +18,90 @@ const SidebarSidecarLayout = ({
   sidebarProps,
   sidebarSlot,
   sidecarSlot,
+  sidebarPropsLevels,
 }: SidebarSidecarLayoutProps) => {
+  const { isDesktop } = useDeviceSize()
+  const numSidebarLevels = sidebarPropsLevels ? sidebarPropsLevels.length : 0
+  const initialSidebarLevel = useMemo(() => {
+    if (numSidebarLevels > 0) {
+      return numSidebarLevels - 1
+    } else {
+      return 0
+    }
+  }, [numSidebarLevels])
+
+  // If `sidebarPropsLevels` is given, set current level to last level.
+  const [currentSidebarLevel, setCurrentSidebarLevel] =
+    useState<number>(initialSidebarLevel)
+
+  /**
+   * @TODO I think this is temporary until all the pages have
+   * `sidebarPropsLevels`. Otherwise, when we switch from something like
+   * /waypoint to /waypoint/docs, the sidebar level that shows is the first one
+   * because this layout component isn't getting re-rendered in a way that
+   * re-initializes `currentSidebarLevel`.
+   */
+  useEffect(() => {
+    setCurrentSidebarLevel(initialSidebarLevel)
+  }, [initialSidebarLevel])
```

Comment:
The intended behavior here would be that the menu opens to the "correct" level (in the case of waypoint/docs, that would be the last level currently), right?

Reply:
Yes! Before I added this effect, the issue I noticed is that the top-level (index 0) nav items were showing when I went from `/waypoint` -> `/waypoint/docs`.